s12/protobuf/proto/validator_helpers.go

```diff
@@ -163,6 +166,28 @@ func IsValidURL(str string, schemes []string, allowFragment bool) (bool, error)
 	return true, nil
 }
 
+// ContainsUrl determines if a string contains a URL anywhere inside it by breaking apart the string at certain
+// characters and inspecting each part.
+//
+// Uses IsValidURL and a subset of schemes to find a URL match.
+func ContainsUrl(fullString string) (bool, error) {
+	stringParts := strings.FieldsFunc(fullString, func(r rune) bool {
+		return r == ' ' || r == ',' || r == '#'
+	})
+	for _, str := range stringParts {
+		// Because the error can be for things like too short a string, we don't want to use it. Instead, ignore it and
+		// continue on to the rest of the string in case we can detect anything further on.
+		//
+		// This is not a perfect solution - nor will it always catch everything - but this should filter out most spam.
+		validUrl, _ := IsValidURL(str, []string{"https", "http", "sftp", "ftp", "file"}, true)
```

Comment:
Let's include `javascript` as a protocol for additional protection against XSS while we're at it.